.github/workflows/docs.yaml

```diff
@@ -0,0 +1,47 @@
+name: OpenAPI
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  build:
+    name: Preview
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: '0'
+    - name: Check whether docs have changed
+      id: checkdocs
+      run: |
+          changed=false
+          git diff --name-only HEAD^ HEAD > files.txt
+          while IFS= read -r file
+          do
+            if [[ $file == openapi/* ]]; then
+              echo "detected openapi spec change"
+              changed=true
+            fi
+          done < files.txt
+
+          if [ $changed == true ]
+          then
+            echo "::set-output name=build_docs::true"
+          else
+            echo "::set-output name=build_docs::false"
+          fi
+    - name: Build the OpenAPI specs
+      if: steps.checkdocs.outputs.build_docs == 'true'
+      uses: acud/openapi-dockerized@v1
```

Comment:
This is a very nice action. I would suggest to paremetrize the endpoint and the bucket name for it. Or just to make it explicitly swarm specific by naming it with something that will make it obvious for that.

Reply:
I agree... I wanted to have it fully parametrized, but it gets a bit more complicated... since I am changing the object ACL, comes the question whether the canned-ACL I'm using also needs to be parametrized, and also the aws CLI tool needs a few tweaks so that it works with DigitalOcean, for example a different endpoint URL:
`aws --endpoint-url https://fra1.digitaloceanspaces.com s3 cp $outfname s3://openapi-specs/`

I am not sure how it will react if a user tries to use it without any concrete endpoint (leading to the endpoint being set as explicitly null for an intended AWS usage). It can further be modularized but then the actual addition of the cli flag to the aws utility must be conditioned too, which is quite ugly. I will add a few additions there, as you suggest :+1:

Reply:
thanks @janos for the great suggestions. I implemented them as suggested :+1: